src/network/protocol/redis/command/network_protocol_redis_command_mget.c

```diff
@@ -0,0 +1,316 @@
+    send_buffer_start = send_buffer;
+    send_buffer_end = send_buffer_start + send_buffer_length;
+
+    send_buffer_start = protocol_redis_writer_write_array(
+            send_buffer_start,
+            send_buffer_end - send_buffer_start,
+            mget_command_context->keys_count);
+
+    if (send_buffer_start == NULL) {
+        LOG_E(TAG, "buffer length incorrectly calculated, not enough space!");
+        goto end;
+    }
+
+    for(int key_index = 0; key_index < mget_command_context->keys_count; key_index++) {
+        storage_db_entry_index_status_t old_status;
+
+        entry_index = storage_db_get_entry_index(
+            db,
+            mget_command_context->keys[key_index]->key,
+            mget_command_context->keys[key_index]->key_length);
+
+        if (likely(entry_index)) {
+                // Try to acquire a reader lock until it's successful or the entry index has been marked as deleted
+                storage_db_entry_index_status_increase_readers_counter(
+                        entry_index,
+                        &old_status);
+
+                // TODO: ??
+                if (unlikely(old_status.deleted)) {
+                    entry_index = NULL;
+                }
+
+                send_buffer_start = protocol_redis_writer_write_argument_blob_start(
+                        send_buffer_start,
+                        send_buffer_length,
+                        false,
+                        (int)entry_index->value_length);
+
+                if (network_send(
+                        channel,
+                        send_buffer,
+                        send_buffer_start - send_buffer) != NETWORK_OP_RESULT_OK) {
+                    goto end;
+                }
+
+                send_buffer_start = send_buffer;
+
+                // Build the chunks for the value
+                for(storage_db_chunk_index_t chunk_index = 0; chunk_index < entry_index->value_chunks_count; chunk_index++) {
+                    storage_db_chunk_info_t *chunk_info = storage_db_entry_value_chunk_get(entry_index, chunk_index);
+                    char *chunk_send_buffer = slab_allocator_mem_alloc(chunk_info->chunk_length);
+
+                    res = storage_db_entry_chunk_read(
+                            db,
+                            chunk_info,
+                            chunk_send_buffer);
+
+                    if (!res) {
+                        LOG_E(
+                                TAG,
+                                "[REDIS][MGET] Critical error, unable to read chunk <%u> long <%u> bytes",
+                                chunk_index,
+                                chunk_info->chunk_length);
+
+                        goto end;
+                    }
+
+                    if (network_send(
+                            channel,
+                            chunk_send_buffer,
+                            chunk_info->chunk_length) != NETWORK_OP_RESULT_OK) {
+                        goto end;
```

Comment:
Better to add a `slab_allocator_mem_free(chunk_send_buffer);` before the `goto end;` to avoid a memory leak